firmware/projects/Demo/CANErrors/inc/app.h

```diff
@@ -0,0 +1,110 @@
+/// @author Luai Bashar
+/// @date 2024-11-10
+/// @brief Functions and types that will be used in CANErrors main
+
+#pragma once
+
+#include "../generated/can/error_can_messages.h"
+#include "shared/comms/can/can_bus.h"
+
+/***************************************************************
+    App-level objects
+***************************************************************/
+
+// Interface to set/send all possible errors for the system
+class ErrorHandler {
+    private:
+        // Object that holds a 64 bit int, each bit representing an error
+        generated::can::TMS_ERROR error_msg{};
+    public:
+        // Sets the error based on the error index given
+        void setError(uint32_t error) {
+            if (error < 0 || error > 63) {
+                std::cerr << "Error index must be between 0 and 63!" << std::endl;
```

Comment:
Up to discussion on how the error handling should be implemented here. I just added a simple log output, but the way this error should be handled should align to how we do it for this whole project.